### PR TITLE
MAISTRA-1057: Add IOR back

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

**Comment by rcernich** [comment]
This will need to be updated after #273 is merged, as the overlays have moved to resources/helm/overlays.  You'll then need to update the v1.1 charts (`make update-1.1-charts`).

File: manifests-servicemesh/1.0.0/servicemeshoperator.v1.0.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -43,7 +43,8 @@ metadata:
                   "autoscaleEnabled": false
                 },
                 "istio-ingressgateway": {
-                  "autoscaleEnabled": false
+                  "autoscaleEnabled": false,
+                  "ior_enabled": false
```
Comment (dgn) [inline manifests-servicemesh/1.0.0/servicemeshoperator.v1.0.0.clusterserviceversion.yaml]: This still has to be removed

File: manifests-maistra/1.0.0/maistraoperator.v1.0.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -43,7 +43,8 @@ metadata:
                   "autoscaleEnabled": false
                 },
                 "istio-ingressgateway": {
-                  "autoscaleEnabled": false
+                  "autoscaleEnabled": false,
+                  "ior_enabled": false
```
Comment (dgn) [inline manifests-maistra/1.0.0/maistraoperator.v1.0.0.clusterserviceversion.yaml]: This still has to be removed

File: deploy/smcp-templates/servicemesh [hunk 3]
```diff
@@ -27,6 +27,9 @@ spec:
       image: citadel-rhel8
     sidecarInjectorWebhook:
       image: sidecar-injector-rhel8
+    gateways:
+      istio-ingressgateway:
+        ior_image: istio-ior-rhel8
```
Comment (dgn) [inline deploy/smcp-templates/servicemesh]: I know this is how it used be handled, but is that really something we should put into `istio-ingressgateway`? Maybe we should rethink this while we're at it.

I'd much prefer if we introduced ior as a component just like other Istio components:
```yaml
spec:
  istio:
    ior:
      enabled: false
      image: istio-ior-rhel8
```

**Review by dgn — COMMENTED** [review]
I think we want to move the manifests into a separate repo- I suppose for that we'll extract them from the `maistra-1.0` branch, so you can likely ignore my comments about the CSVs.

In general I'd prefer it if we would not inject our options into `ingress-gateway` anymore, see my comment below. I think a separate `ior` component block is more in line with the rest of Istio. It's an easy change that only affects the YAMLs, too.

**Review by brian-avery — APPROVED** [review]

**Comment by jwendell** [comment]
> This will need to be updated after #273 is merged, as the overlays have moved to resources/helm/overlays. You'll then need to update the v1.1 charts (`make update-1.1-charts`).

@rcernich PR updated

**Review by rcernich — APPROVED** [review]

**Pull request merged**